File: src/main/java/org/moinex/repository/investment/BondRepository.java
/*
 * Filename: BondRepository.java
 * Created on: January  2, 2026
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 */

package org.moinex.repository.investment;

import java.util.List;
import org.moinex.model.investment.Bond;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface BondRepository extends JpaRepository<Bond, Integer> {
    List<Bond> findByArchivedFalseOrderByNameAsc();

    List<Bond> findByArchivedTrueOrderByNameAsc();

    boolean existsBySymbol(String symbol);
}

File: src/main/kotlin/org/moinex/repository/investment/BondRepository.kt
/*
 * Filename: BondRepository.kt (original filename: BondRepository.java)
 * Created on: January  2, 2026
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 *
 * Migrate to Kotlin on 07/03/2026
 */

package org.moinex.repository.investment

import org.moinex.model.investment.Bond
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface BondRepository : JpaRepository<Bond, Int> {
    fun findByArchivedFalseOrderByNameAsc(): List<Bond>

    fun findByArchivedTrueOrderByNameAsc(): List<Bond>

    fun existsBySymbol(symbol: String): Boolean

    fun existsBySymbolAndIdNot(
        symbol: String,
        int: Int,
    ): Boolean
}
